plugins/available/pip.plugin.bash

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+cite about-plugin
+about-plugin 'pip helper functions'
+
+pip-upgrade-all() {
+	about 'upgrades all user-installed pip packages'
+	group 'pip'
+
+	local outdated_packages="$(pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1)"
+	if [ -z "$outdated_packages" ]; then
+		echo "pip: everything is up to date."
+	else
+		for package in $outdated_packages; do
+			pip install --user -U "$package"
+		done
+	fi
+}
+
+pip-uninstall-all() {
```

Comment:
Similarly, could this have the suffix `-user`?

Reply:
I don't think the `-user` suffix is quite necessary, since pip cannot affect system-installed packages wihout `sudo`, while using `sudo` with pip is discouraged anyway.